### PR TITLE
Enable tests and fix errors

### PR DESCRIPTION
**Pull request by shellyear**
Solving this [issue](https://github.com/kbss-cvut/record-manager-ui/issues/110) 

[1] [info](https://stackoverflow.com/questions/69613533/testing-react-app-with-jest-using-vite-as-a-module-bundler-import-meta-error) 
UPD: Solved by relocation of config folder into src
Screenshot 2024-03-12 at 22 28 27

[2] relates to @kbss-cvut/s-forms dist 
UPD: [%](https://github.com/vitest-dev/vitest/issues/5200)
UPD: Solved by telling eslintPlugin to ignore pattterns (**/dist/*.js)(workaround - should be solved in @kbss-cvut/s-forms, by using named export)
UPD: issue was solved in @kbss-cvut/s-forms; workaround was removed
Screenshot 2024-03-19 at 13 35 13

[3] [%](https://github.com/facebook/react/issues/25446)
Screenshot 2024-03-19 at 16 18 21

[4]  [%](https://vitest.dev/config/#deps-web-transformcss) [%](https://github.com/vitest-dev/vitest/issues/3521#issuecomment-1608527974)
Screenshot 2024-03-21 at 15 34 35

[5] with no s-forms changed, except named imports commit [%](https://docs.triply.cc/yasgui-api/#using-yasgui-in-react)
Screenshot 2024-03-21 at 20 01 54
[6] 
```
SyntaxError: Cannot use import statement outside a module
 ❯ new Script node:vm:94:7

Module /Users/nurbek/Desktop/record-manager-ui/node_modules/react-loader-spinner/dist/esm/index.js:1 seems to be an ES Module but shipped in a CommonJS package. You might want to create an issue to the package "react-loader-spinner" asking them to ship the file in .mjs extension or add "type": "module" in their package.json.

As a temporary workaround you can try to inline the package by updating your config:

// vitest.config.js
export default {
  test: {
    server: {
      deps: {
        inline: [
          "react-loader-spinner"
        ]
      }
    }
  }
}

```

### CODE REVIEW

File: tests/setup.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import enzyme, { shallow, mount } from "enzyme";
-import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
+const Adapter = require("@cfaester/enzyme-adapter-react-18").default;
```
Comment (blcham) [inline tests/setup.js:1]: Let's make issue to remove this deprecated library.

**Comment by blcham** [comment]
@shellyear FYI, most likely you know that checks are failing here.

**Comment by blcham** [comment]
Please spend 15 mins to compare different ways to go. Unless the winner is different from vitest go ahead with the chosen alternative.

**Comment by shellyear** [comment]
@blcham  Current state of tests
Screenshot 2024-03-19 at 2 17 16

**Comment by blcham** [comment]
@shellyear FYI, just watch out, there could be a case (not sure) where the actual logic of the application changed, but tests were not fixed.

**Comment by shellyear** [comment]
@blcham,
The [2] error should be solved in @kbss-cvut/s-forms. 

source code of @kbss-cvut/s-forms (default export):  `import Card from "react-bootstrap/Card"; `
should be named export:` import { Card } from "react-bootstrap";`

I have created a [branch](https://github.com/shellyear/s-forms/tree/fix/imports) for that in my repo, but I think I don't have rights to create a PR for kbss-cvut/s-forms repo

File: tests/setup.js (hunk 1 above)
Comment (shellyear) [reply tests/setup.js:1]: created [issue](https://github.com/kbss-cvut/record-manager-ui/issues/128) for this

**Comment by shellyear** [comment]
[4]
[deps.web.transformCss](https://vitest.dev/config/#deps-web-transformcss)

> _If CSS files are disabled with [css](https://vitest.dev/config/#css) options, this option will just silence ERR_UNKNOWN_FILE_EXTENSION errors._
We don't need to disable css in [css options](https://vitest.dev/config/#css), as some packages used in testing may rely on css

> At the moment, this option only works with [vmThreads](https://vitest.dev/config/#vmthreads) and [vmForks](https://vitest.dev/config/#vmforks) pools.

**Comment by blcham** [comment]
@shellyear 

As I understood that the main problem is related to CSS files, my intuition is as follows:
- we should find out if SForms is the only problem related to CSS (it should be quite easy to remove SForms from the project to test)
- if the problem is SForms only, we need to find out what it means that it is badly bundled  (e.g. compare to something that is not badly bundled :)
- then i guess it is only about finding how to bundle it correctly (e.g. some microbundle param or Rollup more directly)

**Comment by shellyear** [comment]
@blcham I have followed the instructions for [deps.web.transformCSS ](https://vitest.dev/config/#deps-web-transformcss) and used jsdom with vmThreads options, and now as expected I don't have any CSS errors anymore. But I've encountered a new error [6]. The forth error [4] is not relevant anymore

**Comment by shellyear** [comment]
@blcham  can I close this PR ?

**Pull request closed without merging**